includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+		);
+		$body = wp_json_encode( $body );
+		$args = [
+			'method'  => 'POST',
+			'headers' => array(
+				'Content-Type'  => 'application/json',
+				'Authorization' => 'Bearer ' . $api_token,
+				'Cache-Control' => 'no-cache',
+			),
+			'body'    => $body,
+		];
+
+		/**
+		 * Filters the arguments being passed to the override api auth request.
+		 *
+		 * @param array $args Arguments.
+		 * @param string $api_url Auth API URL.
+		 *
+		 * @since 1.3.0
+		 */
+		$args = apply_filters( 'sophi_override_request_args', $args, $api_url );
+
+		if ( function_exists( 'vip_safe_wp_remote_get' ) ) {
+			$result = vip_safe_wp_remote_get( $api_url, '', 3, $timeout, 20, $args );
+		} else {
+			$args['timeout'] = $timeout;
+			$result          = wp_remote_post( $api_url, $args ); // phpcs:ignore
+		}
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		if ( wp_remote_retrieve_response_code( $result ) !== 200 ) {
+			return new \WP_Error( wp_remote_retrieve_response_code( $result ), $result['response']['message'] );
+		}
+
+		if ( 200 === $result['response']['code'] ) {
+			// Update the override entry in the database, so we don't have
+			// to wait for API to update the details at front end.
+			$override_post = [
+				"overridePostID" => $override_post_ID,
+				"position"       => $position,
+				"ruleType"       => 'ban' === $rule_type ? 'out' : $rule_type,
+			];
+			$this->request->get( $page_name, $widget_name, 3, $override_post );
+		}
+
+		return json_decode( wp_remote_retrieve_body( $result ), true );
+	}
+
+	/**
+	 * Get the query params to update the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	public function site_automation_override_params() {
+		$params['ruleType'] = array(
+			'description'       => __( 'The rule of Override.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+		$params['overridePostID'] = array(
+			'description'       => __( 'ID of the post.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'int',
+			'sanitize_callback' => 'absint',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+		$params['pageName'] = array(
+			'description'       => __( 'Name of the page.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+		$params['widgetName'] = array(
+			'description'       => __( 'Name of the widget.', 'sophi-wp' ),
+			'required'          => false,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the get sophi posts query params.
+		 *
+		 * @param array $params Query params.
+		 *
+		 * @since 1.3.0
```

Comment:
Same comment as the other filter, a few changes needed here for our hookdocs

Reply:
https://github.com/globeandmail/sophi-for-wordpress/pull/333/commits/d7f3bb5f76704d21c9cf317832c74ef2d0842ca3